src/App.js

```diff
@@ -33,8 +33,24 @@ function App() {
       })
     .catch((error) => {
       setError(error.message)
-    })}
+    })
+  }
       
+    const changeUrl = () => {
+      const params = new URLSearchParams(window.location.search);
+      params.set('user', inputValue);
+      window.history.replaceState(null, '', `?${params.toString()}`);
+      getRepoByUsername();
+    }
+    
+    useEffect(() => {
+      const params = new URLSearchParams(window.location.search);
+      const usernameFromURL = params.get('user'); 
+      if (usernameFromURL) {
+        setInputValue(usernameFromURL);
```

Comment:
instead of this, use the default useState option

```ts
const [inputValue, setInputValue] = useState(() => {
    const params = new URLSearchParams(window.location.search);
      const usernameFromURL = params.get('query'); 
      return usernameFromURL || ''
       
})
```